src/net_processing.cpp

```diff
@@ -1633,24 +1633,34 @@ void static ProcessGetData(CNode* pfrom, const CChainParams& chainparams, CConnm
             const CInv &inv = *it;
             it++;
 
+            bool too_recent = false;
+            {
+                // Check if the requested transaction is so recent that we're just
+                // about to announce it to the peer; if so, they certainly shouldn't
+                // know we already have it.
```

Comment:
`setInventoryTxToSend` isn't only for initial broadcasts, though. At least the RPC method `sendrawtransaction` can add a transaction to it that has already been relayed previously.

Reply:
That's a good point.

One possibility is making setInventoryTxToSend track a boolean determining if it's a possible re-submission.

Another is just not caring. The entries in setInventoryTxToSend are filted by filterInventoryKnown, so it won't contain anything we know the other party already knows about. Of course, that filter is limited in size, but with 50000 entries I believe it can't expire entries faster than after 47 minutes (50000 invs / 35 (invs / msg) * (2s / msg) = 2857s), long after the entries are gone from the relay map (from which they expire after 15 minutes).